Gems/AtomTressFX/Code/Rendering/HairRenderObject.h

```diff
@@ -0,0 +1,388 @@
+                bool UploadGPUData(const char* name, void* positions, void* tangents);
+
+                //! Preparation of the descriptors table of all the dynamic stream buffers within the class.
+                //! Do not call this method manually as it is called from CreateAndBindGPUResources.
+                //! Can be called from the outside to retrieve the descriptors table (SharedBuffer)
+                static void PrepareSrgDescriptors(
+                    AZStd::vector<SrgBufferDescriptor>& descriptorArray,
+                    int32_t vertexCount, uint32_t strandsCount);
+
+                void PrepareSrgDescriptors(int32_t vertexCount, uint32_t strandsCount)
+                {
+                    PrepareSrgDescriptors(m_dynamicBuffersDescriptors, vertexCount, strandsCount);
+                }
+
+                //! Matching between the buffers Srg and its buffers descriptors, this method fills the Srg with
+                //!  the views of the buffers to be used by the hair instance.
+                //! Do not call this method manually as it is called from CreateAndBindGPUResources.
+                bool BindPerObjectSrgForCompute();
+                bool BindPerObjectSrgForRaster();
+ 
+                Data::Instance<RPI::ShaderResourceGroup> GetSimSrgForCompute() { return m_simSrgForCompute; }
+                Data::Instance<RPI::ShaderResourceGroup> GetSimSrgForRaster() { return m_simSrgForRaster; }
+
+                bool IsInitialized() { return m_initialized;  }
+
+            private:
+                //! The descriptors required to allocate and associate the dynamic buffers with the SRGs
+                //! Each descriptor also contains the byte offsets of the sub-buffers in the global dynamic
+                //!  array for the data copy.
+                AZStd::vector<SrgBufferDescriptor> m_dynamicBuffersDescriptors;
+
+                //! The following dynamic buffer views are views 'sub-buffers' located within a global large
+                //!  dynamic buffer exposed and connected as an attachment between the passes and therefore
+                //!  creates both dependency order between passes execution and sync point barrier.
+                //! This indirectly forces the sync to be applied to all 'sub-buffers' used by each of the
+                //!  HairObjects / HairDispatches and therefore allows us to change their data in the shader
+                //!  between passes.
+                AZStd::vector<Data::Instance<RHI::BufferView>> m_dynamicBuffersViews;   // RW used for the Compute
+                AZStd::vector<Data::Instance<RHI::BufferView>> m_readBuffersViews;      // Read only used for the Raster fill
+
+                //! The following vector is required in order to keep the allocators 'alive' or
+                //!  else they are cleared from the buffer via the reference mechanism.
+                AZStd::vector<Data::Instance<SharedBufferAllocation>> m_dynamicViewAllocators;
```

Comment:
the difference is in understanding and while you are right in the fact that they do not do the actual allocation but hold the allocation data, I'd argue it is somewhat clearer this way. 
Since it's a nit I'll close it and you can decide if you want to re-open.